internal/cli/errors.go

```diff
@@ -0,0 +1,64 @@
+package cli
+
+import (
+	"fmt"
+)
+
+func NewErr(message string) Err {
+	return Err{message: message}
+}
+
+func NewErrw(message string, err error) Err {
+	return Err{message: message, cause: err}
+}
+
+func NewPriviligedErr(message string, err error) PriviligedErr {
+	return PriviligedErr{NewErrw(message, err)}
+}
+
+type WrappedErr interface {
+	Unwrap() error
+}
+
+type Err struct {
+	WrappedErr
+	message string
+	cause   error
+}
+
+func (err Err) Error() string { return err.message }
+
+// Unwrap unwraps the first non-CLI error as the root cause
+func (err Err) Unwrap() error { return findRootCause(err.cause) }
+
+func (err Err) String() string {
+	if err.cause == nil {
+		return err.message
+	}
+
+	var cause string
+	switch c := err.cause.(type) {
+	case Err:
+		cause = c.String()
+	default:
+		cause = c.Error()
+	}
+	return fmt.Sprintf("%s: %s", err.message, cause)
+}
+
+type PriviligedErr struct {
+	Err
+}
+
+func (err PriviligedErr) Error() string {
+	return fmt.Sprintf("%s: %s", err.message, err.Unwrap().Error())
+}
+
+func findRootCause(err error) error {
+	switch e := err.(type) {
+	case WrappedErr:
+		return findRootCause(e.Unwrap())
```

Comment:
I think this may change as we move forward with implementation so happy to punt on it. I am curious on if we do our implementation right we won't have to wrap many errors, letting us remove this in the future.

Reply:
👍 ... error handling in general I'm leaving very naive right now, but there's going to be _some way_ these things are captured so we can present nicely formatted error messages back (with help, tips etc) to the user without the developer needing to know/do much on that front